bindings/python/pyproject.toml

```diff
@@ -0,0 +1,3 @@
+[build-system]
+requires = ["setuptools>=41", "cython>=0.29<1", "pyarrow>=3,<4", "wheel"]
+build-backend = "setuptools.build_meta"
```

Comment:
Why are we adding `pyproject.toml`? My main apprehension is the duplicate cython/pyarrow versions requires.

Reply:
setuptools reads pyproject.toml so we can now remove `setup_requires` (which was useless anyway) and `python setup.py bdist_wheel` will automatically install the build deps.